pkg/pod/status.go

```diff
@@ -170,7 +170,15 @@ func setTaskRunStatusBasedOnStepStatus(logger *zap.SugaredLogger, stepStatuses [
 					logger.Errorf("error extracting the exit code of step %q in taskrun %q: %v", s.Name, tr.Name, err)
 					merr = multierror.Append(merr, err)
 				}
-				taskResults, pipelineResourceResults, filteredResults := filterResultsAndResources(results)
+				// collect results from taskspec
+				specResults := []v1beta1.TaskResult{}
+				if tr.Spec.TaskSpec != nil {
+					specResults = append(specResults, tr.Spec.TaskSpec.Results...)
```

Comment:
Just wondering if this works if `tr.Spec.TaskSpec.Results` is nil?

Reply:
yes, tr.Spec.TaskSpec.Results is a list, so if it is an empty list it should be fine.

Reply:
SGTM!